### ISSUE TITLE
Implement High contrast theme

### ISSUE THREAD

**Issue by veselints** [telerik/kendo-themes#1629]
### Implement High contrast theme
The presence of a High contrast theme is an essential requirement for the accessibility compliance of all components relying on the Kendo SASS themes. Such a theme is extremely important for all Kendo products because it would allow us to:
- Cover the WCAG 2.1 AA standard requirements for contrast ratio between text and background colors;
- Provide easy-to-find examples, where customers may test the compliance of our components (within the demo/docs applications for each suite);
- Advertise the compliance of the components and the suites as a whole supporting our message with live examples;
- Entirely remove (or hide) the LESS based themes from the Kendo, MVC and Core demos applications, so that we put emphasis on the SASS based themes and the use of the SASS Theme builder;

**Comment by veselints** [comment]
I know that there is an idea of limiting the theme chooser in the demos app to offer only the SASS themes. That, I am afraid, won't be possible until we have the High contrast in our offering. It will be difficult to advertise our product as WCAG compliant without having an easily accessible online demo where the compliance could be tested.

**Comment by joneff** [comment]
Part of #994

**Comment by nenchef** [comment]
Besides the advantages that Veso listed above, I can add the fact that our competitors have such a high contrast theme. I haven't dig deeper into details, however, and the exact implementation at their end.
It's a good thing that we raise the discussion here. The fact that we are talking about this theme for a **couple of years now**, makes the current state a bit obscure. 
@joneff  could you please shed some light on the matter?- current state and our plans. As this is a common themе, it would be valuable to hear opinions from the other teams as well.

cc @Raisolution  @tapopov @eeasss @hadjieva @Xizario @VicTachev

**Comment by VicTachev** [comment]
I agree that a high contrast theme would be a good enhancement to our current offering.

**Comment by nenchef** [comment]
@Raisolution @tapopov @Xizario
Guys, could you please share your thoughts on the matter? How is the accessibility topic covered in Ang & React suites? Do we have any official statement or urgency of such a high-contrast theme?

**Comment by Raisolution** [comment]
The accessibility compliance is documented among the common features of the components with information about the standards we are aiming to support (Section 508, WCAG 2.1)
https://www.telerik.com/kendo-angular-ui/components/accessibility/

The section contains a table with this information per component and links to the detailed component sections and their keyboard support documentation.
https://www.telerik.com/kendo-angular-ui/components/accessibility/accessibility-compliance/

I am not aware of any peek in the requests about high contrast theme, it has been occasionally brought up.

I have a question - are we talking about a **new theme** or a **new color swatch**? If that is a theme, what else will be different except colors? If this is a swatch, we can refer it as part of telerik/kendo-ux-private#72 and work towards **showcasing all our swatches** more visibly. This will allow us to also showcase the **black swatches** too, as **"dark mode"** is also commonly requested feature for our components.

**Comment by veselints** [comment]
@Raisolution, thank you for your clarification on the Angular components.

As per the question swatch vs new theme, I believe that even a High contrast swatch will completely fulfill our needs.

I will bring a small example of why we need that swatch/theme even in the Angular context. If you review the colors for the TabStrip component (https://www.telerik.com/kendo-angular-ui/components/layout/tabstrip/), the contrast ratio of the non-active tabs will fail the WCAG AA criteria for the Default and for the Bootstrap theme. The component, however, is marked as WCAG AAA compliant (https://www.telerik.com/kendo-angular-ui/components/accessibility/accessibility-compliance/), while that is the case only with the Material theme.

After some random tests on the Kendo jQuery widgets with the SASS Material theme, I did not find a failing contrast ratio with that theme. We could perform careful testing with that to check whether it could work as a "high contrast" theme in the SASS bundle. @Dimitar-Goshev, what do you think?

**Comment by eeasss** [comment]
@nenchef @veselints the actual work on designing high-contrast swatch should start with our UI team in Kendo UX Private after we have clarified what exactly do we want to achieve. In order to actually start working on this we should have a product manager buy in and it should became part of our release planning.

The reason item got postponed for the last few years was lack of resources and low prioritization (in favor of other accessibility items). 

I also have distant memories that @tsvetomir had some comments on the viability of high-contrast theme but I could be wrong as well. :) 

I suggest that as a next step we bring up this item for consideration for the next planning.

**Comment by eeasss** [comment]
@veselints also high-contrast swatch already exists on the theme-builder and is applicable for Kendo UI. Have you tested it? It was part of a thorough research that Lili did back in the days. If all is good with the swatch we can do design revision and then think about it makes sense to be released as a separate them or instead better showcase the swatch it in the demos, as @Raisolution mentioned.

**Comment by eeasss** [comment]
![image](https://user-images.githubusercontent.com/598514/85459413-d0c72000-b5aa-11ea-887b-0bb5dcb4f9bc.png)

**Comment by tsvetomir** [comment]
A swatch, or a set of swatches, looks like the right way to approach this. The ThemeBuilder could calculate the contrast ratio for the key colors to help with building and customizing such swatches.

Note that the contrast of black-on-lipstick text is too low (1.53) in the current swatch:
![image](https://user-images.githubusercontent.com/247917/85502815-d21b3b80-b5f0-11ea-933f-bb82e5ceafed.png)

**Comment by nenchef** [comment]
An idea of a High Contrast theme was present in the planning for several iterations, but we never got into it. Indeed, a swatch was identified as a short term solution, but we agreed that if we put any effort on that front, it won't worth it creating a swatch, but rather implement the Theme itself. Indeed there are some metrics that are not correct with the current high contrast swatch. 
On the other hand, having a standalone theme will allow us to include it in the ThemeChooser dropdown on the demos/docs sites, and thus, emphasize the accessibility coverage of the products.

**Comment by veselints** [comment]
@eeasss unfortunately, as @tsvetomir has pointed out, the High Contrast swatch in its current state does not cover the WCAG AA standard. Therefore we definitely need to either revise that swatch or offer a different solution.

@nenchef as per the ThemeChoser, I believe that with some effort, we could also place there a specific swatch of a given theme. Therefore, I do not think creating a separate High Contrast theme is a must. Moreover, having a swatch for each theme (instead of a dedicated theme) will make all the three SASS themes WCAG ready by keeping their dimensions and typography and modifying only the colors used. 

One thing that we should consider and review is the compliance of the Material SASS theme itself. It may turn out that it covers the WCAG AA standard or it may need minor improvements to cover it. My suggestion for an initial action is to test Material theme compatibility in its current state.

**Comment by eeasss** [comment]
To answer whether we need a new theme or now let's think about our motivation to create themes in the first place (and please avoid thinking by analogy to our previous approach of creating 20+ "skins"):
1. We have to support new design system
2. That have different metrics
3. And likely different interactions and in extreme cases extra dom elements (ripple effect for example)

The high-contrast theme need not to have different metrics, can easily use the same design rules as the default theme and it is likely that no rendering changes will be needed. In this sense - it fits our existing definition of swatches. 

Since I don't see evidence that we need new theme my proposed steps are 1) we figure out to showcase swatches in the demos (important item) 2) we evaluate whether we should build the high-contrast swatch on the new kendo design system than @tyanko  is creating or on the existing default theme or both 3)  we revise the existing high-contrast swatch and redesign it as needed 4) only then the front-end team applies the needed fixes. 

@veselints can you please elaborate what is the connection between the material theme and the high-contrast swatch? Material theme WCAG support seems to be a ifferent topic.

**Comment by veselints** [comment]
@eeasss, yes, I will try to explain briefly, why I am bringing the Material theme in the discussion.

The reason for having a High Contrast swatch in the first place is to cover the contrast ratio requirements of the WCAG AA (and where possible, the AAA) standard. If one of the already existing themes covers those requirements anyways, we can test our components against that theme (we won't need the dedicated swatch for our test environment). Also, when advertising the accessibility of our components we can clearly state that the contrast ratio standard is covered by that theme (just like we currently do with the High Contrast LESS theme in the jQuery suite). Moreover, when the customer can easily run an example using that theme (from our demos), she can also run a quick test to check the WCAG and Section 508 compatibility of a given component. Passing such a test would be a great selling point for any of the suites.

Having all that said if the Material theme covers the contrast ratio requirements, it means we already have a theme that fulfills the primary purpose of a High Contrast swatch. That does not mean we should not improve the current High Contrast swatch. Nevertheless, it gives us confidence in advertising our components. As per the jQuery widgets, it also gives us the freedom to entirely drop the LESS based themes from our demos site and possibly from the accessibility test environment (Compliance Sheriff tests).

**Comment by eeasss** [comment]
@veselints Ok, thanks for the clarification. But what makes you think that the material theme would cover those? Why not the bootstrap or the default?

**Comment by veselints** [comment]
@eeasss I have tested several widgets in Material theme and those have passed the contrast ratio requirements. That is not the case for the Default and Bootstrap theme. In my first comment above I have mentioned that the TabStrip fails the validation while using those two themes.

As per the tool, I use the WAVE extension for Chrome: https://wave.webaim.org/extension/

**Comment by joneff** [comment]
Before we get to the high contrast theme, we must first ensure the components are reusable enough.

* As far as pivot grid is concerned, those button like indicators are not styled the best way possible, hence the issues
* the same could be said about the footer / aggregate. Speaking of which, Ivelina is already overhauling the grid design, so this part can be somewhat mitigated.

So as we are moving forward with migrating the less themes and v5, we are clearing all the issues that stand in the way of a truly high contrast theme.

**Comment by eeasss** [comment]
@joneff do you really think we should go out with a full blow theme or a swatch configuration is more appropriate? Personally I don't see added value in going for the high-contrast theme approach unless we change the metrics. 
To put it more clearly, I think the only objective reason to push forward theme vs the swatch story is a change in the metrics.

What do you think?

**Comment by tapopov** [comment]
Going forward with the swatch first have its benefits. We get immediate results with much less effort AND we can reuse the palette later, should we go for a full-blown high-contrast theme.

**Comment by joneff** [comment]
The theme, while harder will provide for better accessibility.

A swatch can only get us somewhat far. For the time being, we are going to fix the issues with the high contrast swatch and probably make it part of the themes.

**Comment by eeasss** [comment]
Makes sense. Let's also be real about our capacity right now. We simply don't have resources to create high contrast theme right now, but could use the swatch until we get there. //cc @bergenhem

**Comment by bergenhem** [comment]
Just to add my $0.02: I think that starting with a swatch to begin with as a "testing the waters" kind of approach makes sense.

To me the full theme would be us doing some thought leadership and showcase our full commitment to accessibility. While creating this theme is good for us to do (and something that I want us to do), I don't think we will move away from mostly getting comments on our HTML rendering and approach to certain components where the theme is not the culprit.

**Comment by eeasss** [comment]
Part of: https://github.com/telerik/kendo-okrs/issues/27

**Comment by eeasss** [comment]
Update: the existing high-contrast swatch will be removed from the themebuilder, as a part of the revamp we are doing for the launch for Figma components. 

When we create new swatch, as a part of the referenced issue from kendo-ux-priavate we will include it once again into the TB.

**Comment by jrmcdona** [comment]
Is there a high contrast theme for Kendo react?  I tried to simply override the styles in high contrast mode and it doesn't work.

Here is what I tried.  Please note the override works in non high contrast mode but as soon as I switch Windows to high contrast it does not work.   I am trying to fix accessibility bug required a Microsoft.

```
@media screen and (prefers-color-scheme: dark), screen and (prefers-contrast: more) {
  /* Dark Mode styles go here */
  .k-list .k-item:hover {
    background-color: yellow !important;
  }
  .k-list .k-item:hover.k-state-selected {
    color: white;
    background-color: yellow !important;
  }
}

@media screen and (prefers-contrast: more) {
  /* Increased Contrast Mode styles go here */
  .k-list .k-item:hover {
    background-color: yellow !important;
  }
  .k-list .k-item:hover.k-state-selected {
    color: white;
    background-color: yellow !important;
  }
}

    // High Contrast Mode
    // Use native browser styles
    @media screen and (-ms-high-contrast: active) {
      /* make it red instead (with with same width and style) */
      .k-list .k-item:hover {
        background-color: yellow !important;
      }
      .k-list .k-item:hover.k-state-selected {
        color: white;
        background-color: yellow !important;
      }
    }
```
